### ISSUE TITLE
Bad block during synchronization of Rinkeby archive node

### ISSUE THREAD

**Issue by bertux** [ethereum/go-ethereum#20907]
Hi there,

#### System information

Geth version: `geth version`
OS & Version: Linux Ubuntu 18.04
Commit hash : running from stable release
original ExecStart command in systemctl taken from BlockScout : `/usr/bin/geth --rpc --rpcaddr 0.0.0.0 --port 30303 --rpcport 8545 --rpcapi debug,net,eth,shh,web3,txpool --wsapi "eth,net,web3,network,debug,txpool" --ws --wsaddr 0.0.0.0 --wsport 8546 --wsorigins "*" --rinkeby --datadir=/home/eth/rinkeby --syncmode=full --gcmode=archive --rpcvhosts=*`

current ExecStart command in systemctl taken from rinkeby.io : `/usr/bin/geth --networkid=4 --datadir=/home/eth/rinkeby --cache=4096 --syncmode=full --gcmode=archive --ethstats='strat0s:Respect my authoritah!@stats.rinkeby.io' --bootnodes=enode://a24ac7c5484ef4ed0c5eb2d36620ba4e4aa13b8c84684e1b4aab0cebea2ae45cb4d375b77eab56516d34bfbd3c1a833fc51296ff084b770b94fb9028c4d25ccf@52.169.42.101:30303`

#### Expected behaviour
To finish well the archive node synchronization to be able to run BlockScout with it.

#### Actual behaviour
Looping on this bad block
On the same machine with the same geth I was able to synchronize Rinkeby in syncmode fast but archive node is needed by BlockScout because it tries to index all the historical states.

#### Steps to reproduce the behaviour
just run the same first command

#### Backtrace

````
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: ERROR[04-09|17:04:40.397]
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: ########## BAD BLOCK #########
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: Chain config: {ChainID: 4 Homestead: 1 DAO: <nil> DAOSupport: false EIP150: 2 EIP155: 3 EIP158: 3 Byzantium: 1035301 Constantinople: 3660663 Petersburg: 4321234 Istanbul: 5435345, Muir Glacier: <nil>, Engine: clique}
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: Number: 3177502
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: Hash: 0xb24d0abbac0a117f820cb2a000d769283cc53f1f4b04c9747a35bbd82756f8be
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          0: cumulative: 58894 gas: 58894 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x7cc3a2069d801eb6a0370f3791d8146da4e9b2461c3b091cb11988283a104898 logs: [0xc007c7f970] bloom: 00000000000000000000000000000000000000000000000000800000000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          1: cumulative: 246436 gas: 187542 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x602aed3e8d92c0aed5478c5d41bdb7bda12d1a96bacbd0ddaa1e675c1ccba3c9 logs: [0xc007c7fa20] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          2: cumulative: 337959 gas: 91523 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x60a99fc977f693f0c8e66150130a07a72ad996a65f63b5cc62cbc71c5f07dbd0 logs: [0xc007c7fad0] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          3: cumulative: 460434 gas: 122475 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xfd1c90dbfdc1ddb6941545b0a1e3ff24c5c043b08978a0dbea532403a67876a1 logs: [0xc007c7fb80] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          4: cumulative: 551957 gas: 91523 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xe8336661fb0e0301a96925defc84da9aa17b11ab7e4e40395b07eddcab76041a logs: [0xc007c7fc30] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          5: cumulative: 643480 gas: 91523 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x98431b1c566b66d8fa2df86b33b7a5a070d697bb7a6747c2572b4ee7e8019175 logs: [0xc007c7fce0] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          6: cumulative: 708866 gas: 65386 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x43b7cf97ff67ad0a3bdce0266317405683fb9d537481eeebbaa8e54013088213 logs: [0xc007c7fd90] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          7: cumulative: 800389 gas: 91523 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x83bf6f95d287972f20a7bd8df072abd4d161ef62f3396ee0f72e008b4ce37906 logs: [0xc007c7fe40] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          8: cumulative: 891912 gas: 91523 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xc4213b7e568838a6a99564cb63d660a03b1c2c80dec86f5124ebe0d780dc1e81 logs: [0xc007c7fef0] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          9: cumulative: 957298 gas: 65386 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xbe2af7b869d83afa703451b00619d2ca4fd5a8e57d1ec8a54987b3e08070911e logs: [0xc00abc0000] bloom: 00000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          10: cumulative: 991718 gas: 34420 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x5c259d7f10469a64ed06468d51417e928f5571b6aa0eb87f4205af6c12de3d35 logs: [0xc00abc00b0] bloom: 00000000000000080000000000000000000000000000000000000000000000000000000000000000000000000400000000000000000008000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000800000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          11: cumulative: 1069029 gas: 77311 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x6761e100eda29bde324a749b426bacb3ac790e91d3e805208252abc5ee69dcc5 logs: [0xc00abc0160] bloom: 00000000000000000000000000000000000000000000000000000000000000000000200040000000000000000000000000000000000000200000000000000040000000000040000000000000000000000000000000000000000000000008000000000810000000000000000000000000000000000000000000200000000100000000000000000000000000000800000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000100000020000000000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          12: cumulative: 2075122 gas: 1006093 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x2ec3bad9c2d532bcf9420a94aac75e66f4789429633eb1e35e7e6d1d8c9f7bc2 logs: [0xc00abc0210 0xc00abc02c0 0xc00abc0370 0xc00abc0420 0xc00abc04d0 0xc00abc0580 0xc00abc0630 0xc00abc06e0 0xc00abc0790 0xc00abc0840] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000008000000000000000000001000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          13: cumulative: 2792143 gas: 717021 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xb5eb5139f9e8ca699125f0ea14a44239701fc1d176b48a9ed3197fcb73767828 logs: [0xc00abc08f0 0xc00abc09a0 0xc00abc0a50 0xc00abc0b00 0xc00abc0bb0 0xc00abc0c60 0xc00abc0d10] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000008000000000000000000001000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          14: cumulative: 3178260 gas: 386117 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x80523e0f8d73d4929a744689f3e6585d39e8aa07b561a57ff80fec9f73aacde6 logs: [0xc00abc0dc0 0xc00abc0e70 0xc00abc0f20 0xc00abc0fd0 0xc00abc1080] bloom: 00000000100000000000000000010200000000000004000000000000800000000000000000000000000100000000000000008400000000080000000000000000000000000000000000000008000008000000000004000800000000200000000000000000040000000001000000000000000011000000000000000010000000000000000000000000000000000800000000000000000000000008000020000000000004000100000000000000080000008000000000000100000000100000000000000002000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          15: cumulative: 3298492 gas: 120232 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x32a3f76f48f3b5598ff9dd8f8285276a4f61d3d667c74e79046b210a318fdb15 logs: [0xc00abc1130 0xc00abc11e0] bloom: 00000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000100000000020000000000000000000000000000000000000000000008080000000000000000000000000000000000000000000010080000000000000000000000000000000000020000000010000000000000000000000000000000000000000000000000000000000000000000000000000000401000000000000040000000000000000000000010000000000000000000000002000000000000040000000000000000000000000000000000000000000000000000000000000004000000000000000100010000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          16: cumulative: 3393223 gas: 94731 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x97daf96935fed91b4397512e4743198c1fc70e3a28b5a5c5e4e063f4e2300998 logs: [0xc00abc1290] bloom: 00000400000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000000020000000000000000000800000020000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000040000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          17: cumulative: 5033172 gas: 1639949 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x7852ac94ed3f1b7606c366f7eb52a286b76ed1e33e0e0ebc7cfc8093b1b2f89c logs: [0xc00abc1340 0xc00abc13f0] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000c00000400020000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000800000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000400000000000000000000000008000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000020 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          18: cumulative: 5332425 gas: 299253 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0x3becdc931d6872583e60d3175358874c7c675e34d4459b58de4fbd5e597e6070 logs: [0xc00abc14a0 0xc00abc1550] bloom: 00000000200000000000000000000000000000000000000800000000000000000000000000000000000000202200000000000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000000000000000000040000000000000000000000008000000000000000000000000000000000000200000000200000000000000000000000020000000000400000000000000000000000100000000000000000000000000000000010000020000000000000000000000080000000000000000000000000000000000000000400000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:          19: cumulative: 5353425 gas: 21000 contract: 0x0000000000000000000000000000000000000000 status: 1 tx: 0xffd16bf6b19c65b4a91a1dfa80b4dc1e6733b9c001bf8d7c7186c0880e991d0a logs: [] bloom: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 state:
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: Error: invalid gas used (remote: 6508425 local: 5353425)
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: ##############################
Apr 09 17:04:40 vm.strat0s.dev geth[9456]:  
Apr 09 17:04:40 vm.strat0s.dev geth[9456]: WARN [04-09|17:04:40.398] Synchronisation failed, dropping peer    peer=56483dbe3073c572 err="retrieved hash chain is invalid"
````

**Comment by bertux** [comment]
I try to compare it with https://rinkeby.etherscan.io/block/3177502 where I see also 20 transactions for this block.

**Comment by holiman** [comment]
> I really need to remove this block from my chaindata. Should I use get export for the first 3177501 blocks, remove the chaindata and import this export ?

The probem isn't the block, but the state. So yes, you can export the blocks as RLP, and then do a import from RLP (after removing the chaindata). 

I don't understand why you are using two different commands? Did you do a clean shutdown between them? It should be sufficient to do only one command. Note though, that if you do not specify `--rinkeby` OR do a `geth init`, it won't be correctly configured for rinkeby. In your case, it's fine though, because the first command fixed that for you.

**Comment by bertux** [comment]
I've switched to full node instead of archive node since that issue.
Thank you for confirming my export id.
I have started by using in my systemctl service the command recommended by blockscout and then because of this issue I've replaced it with the command recommended on rinkeby.io and I've well done clean shutdown. But as you said the problem is in the internal state of my chaindata, removing it and syncing in full mode instead of archive mode I haven't this issue anymore so I'm closing it.
